torch/utils/_sympy/functions.py

```diff
@@ -76,9 +89,9 @@
 ]
 
 
-def _keep_float(f):
+def _keep_float(f: Callable[..., T]) -> Callable[..., Union[T, sympy.Float]]:
```

Comment:
If T supports float already, what does it mean for the return type of the callable to be Union(T, sympy.Float)?